src/interface/src/styleguide/sidebar-name-input/sidebar-name-input.component.ts

```diff
@@ -0,0 +1,92 @@
+// handle errors
+// emit change event
+
+import {
+  Component,
+  Input,
+  Output,
+  EventEmitter,
+  OnDestroy,
+  OnInit,
+} from '@angular/core';
+import { CommonModule, NgIf } from '@angular/common';
+import { FormsModule } from '@angular/forms'; // Import FormsModule
+import { InputFieldComponent } from '@styleguide';
+import {
+  debounceTime,
+  distinctUntilChanged,
+  Subject,
+  Observable,
+  BehaviorSubject,
+} from 'rxjs';
+import { InputDirective } from '../input/input.directive';
+import { MatIconModule } from '@angular/material/icon';
+import { MatMenuModule } from '@angular/material/menu';
+import { MatButtonModule } from '@angular/material/button';
+import { MatProgressSpinnerModule } from '@angular/material/progress-spinner';
+import { UntilDestroy, untilDestroyed } from '@ngneat/until-destroy';
+
+/**
+ * Component for setting name
+ */
+@UntilDestroy()
+@Component({
+  selector: 'sg-sidebar-name-input',
+  standalone: true,
+  imports: [
+    CommonModule,
+    FormsModule,
+    InputFieldComponent,
+    InputDirective,
+    MatButtonModule,
+    MatIconModule,
+    MatMenuModule,
+    MatProgressSpinnerModule,
+    NgIf,
+  ],
+  templateUrl: './sidebar-name-input.component.html',
+  styleUrl: './sidebar-name-input.component.scss',
+})
+export class SidebarNameInputComponent implements OnInit, OnDestroy {
```

Comment:
I'm not sure about this component name, but I wanted to convey that this is a generic input component.

Reply:
Do you think this is going to be a somewhat shared component? Looks like its a field + label + debounce time for emitting changes . Maybe something related to that debounce saving logic instead of sidebar?

Reply:
Sure. I wanted to make it clear that this was a component designed to be added to the left side content, since we have a few examples of that, but I suppose it can be more flexible than this.